community/cypher/cypher-compiler-3.2/src/main/scala/org/neo4j/cypher/internal/compiler/v3_2/codegen/ir/aggregation/AggregateExpression.scala

```diff
@@ -48,12 +48,16 @@ abstract class BaseAggregateExpression(expression: CodeGenExpression, distinct:
     expression match {
       case NodeExpression(v) => primitiveIfNot(v, structure)(block(_))
       case RelationshipExpression(v) => primitiveIfNot(v, structure)(block(_))
-      case _ =>
+      case expr =>
         val tmpName = context.namer.newVarName()
         structure.assign(tmpName, expression.codeGenType, expression.generateExpression(structure))
-        structure.ifNonNullStatement(structure.loadVariable(tmpName)) { body =>
+        val perhapsCheckForNotNullStatement: ((MethodStructure[E]) => Unit) => Unit = expr match {
```

Comment:
There is already a standard way for checking if an CodeGenExpression can be null: `nullable`.
For example,
```
        val perhapsCheckForNotNullStatement: ((MethodStructure[E]) => Unit) => Unit =
          if (expr.nullable)
            l1.ifNonNullStatement(l1.loadVariable(tmpName))
          else
            (f) => f(l1)
```